source/hic_hal/stm32/stm32h743ii/extended_features/source/DAP_vendor_ex.c

```diff
@@ -283,6 +290,82 @@ uint32_t DAP_ProcessVendorCommandEx(const uint8_t *request, uint8_t *response) {
 
         break;
     }
+    case ID_DAP_VendorEx40_MEASURE_POWER:
+    {
+        bool ret = udb_power_measurement_measure();
+
+        if (ret)
+        {
+            *response = DAP_OK;
+        }
+        else
+        {
+            *response = DAP_ERROR;
+        }
+        num += 1;
+        break;
+    }
+    case ID_DAP_VendorEx41_READ_POWER:
+    {
+        uint8_t target_type = *request++;
```

Comment:
Can you move the processing for this command into a separate function? It's kind of long, and it's a little easier to understand when the targets of this sort of goto error handling are at the end of a function body rather than just at the end of the current block.